### ISSUE TITLE
IndexError with invalid `map()`

### ISSUE THREAD

**Issue by blueyed** [Kuniwak/vint#256]
The following `t.vim` triggers an IndexError:, but should rather point out that `expr2` is missing.

```vim
call string(map(foo), 'v:val')
```

```
Traceback (most recent call last):
  File "…/src/neomake/.venv/bin/vint", line 11, in <module>
    load_entry_point('vim-vint', 'console_scripts', 'vint')()
  File "…/Vcs/vint/vint/__init__.py", line 11, in main
    init_cli()
  File "…/Vcs/vint/vint/bootstrap.py", line 22, in init_cli
    cli.start()
  File "…/Vcs/vint/vint/linting/cli.py", line 27, in start
    violations = self._lint_all(env, config_dict)
  File "…/Vcs/vint/vint/linting/cli.py", line 120, in _lint_all
    violations += linter.lint_file(file_path)
  File "…/Vcs/vint/vint/linting/linter.py", line 107, in lint_file
    root_ast = self._parser.parse_file(path)
  File "…/Vcs/vint/vint/ast/parsing.py", line 42, in parse_file
    return self.parse(decoded_and_lf_normalized)
  File "…/Vcs/vint/vint/ast/parsing.py", line 30, in parse
    plugin.process(ast)
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/__init__.py", line 32, in process
    processed_ast = self._ref_tester.process(ast)
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/reference_reachability_tester.py", line 43, in process
    scope_linker.process(ast)
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/scope_linker.py", line 312, in process
    attached_ast = id_classifier.attach_identifier_attributes(ast)
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/identifier_classifier.py", line 126, in attach_identifier_attributes
    map_and_filter_parser.process(ast_with_parsed_redir)
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/map_and_filter_parser.py", line 44, in process
    traverse(ast, on_enter=enter_handler)
  File "…/Vcs/vint/vint/ast/traversing.py", line 244, in traverse
    node[prop_name])
  File "…/Vcs/vint/vint/ast/traversing.py", line 9, in for_each
    call_if_def(func, node)
  File "…/Vcs/vint/vint/ast/traversing.py", line 23, in call_if_def
    func(node)
  File "…/Vcs/vint/vint/ast/traversing.py", line 243, in <lambda>
    accessor_func(lambda child_node: traverse(child_node, on_enter, on_leave),
  File "…/Vcs/vint/vint/ast/traversing.py", line 244, in traverse
    node[prop_name])
  File "…/Vcs/vint/vint/ast/traversing.py", line 23, in call_if_def
    func(node)
  File "…/Vcs/vint/vint/ast/traversing.py", line 243, in <lambda>
    accessor_func(lambda child_node: traverse(child_node, on_enter, on_leave),
  File "…/Vcs/vint/vint/ast/traversing.py", line 234, in traverse
    should_traverse_children = on_enter(node) is not SKIP_CHILDREN
  File "…/Vcs/vint/vint/ast/plugin/scope_plugin/map_and_filter_parser.py", line 34, in enter_handler
    string_expr_node = node['rlist'][1]
IndexError: list index out of range
```

Printing `node` there gives:

> {'type': 78, 'pos': {'i': 8, 'lnum': 1, 'col': 9}, 'left': {'type': 86, 'pos': {'i': 5, 'lnum': 1, 'col': 6}, 'value': 'map'}, 'rlist': [{'type': 86, 'pos': {'i': 9, 'lnum': 1, 'col': 10}, 'value': 'foo'}]}

**Comment by Kuniwak** [comment]
This bug was fixed at v0.3.18. Please try it.

**Comment by blueyed** [comment]
Thank you, works!
